java/fury-core/src/main/java/org/apache/fury/type/Descriptor.java

```diff
@@ -143,14 +145,28 @@ private Descriptor(
     this.furyField = this.field == null ? null : this.field.getAnnotation(FuryField.class);
   }
 
+  public Descriptor(DescriptorBuilder builder) {
+    this(
+        builder.typeRef,
+        builder.typeName,
+        builder.name,
+        builder.modifier,
+        builder.declaringClass,
+        builder.field,
+        builder.readMethod,
+        builder.writeMethod);
+    this.nullable = builder.nullable;
+    this.trackingRef = builder.trackingRef;
+    this.type = builder.type;
+    this.furyField = builder.furyField;
+  }
+
   public Descriptor copy(Method readMethod, Method writeMethod) {
```

Comment:
Could we also add a `copyBuilder()` method in Descriptor to simplify the usage. And could we add a test?